infra/azure/modules/batch/main.tf

```diff
@@ -215,6 +247,42 @@ resource "azurerm_role_assignment" "test_test_container_contributor" {
   principal_id         = module.test_sp.principal_id
 }
 
+resource "azurerm_role_assignment" "test_registry_viewer" {
```

Comment:
maybe call this `test_sp_registry_viewer`? `test_registry_viewer` seems too much like test is an adjective.